tracer/build/_build/Build.GitHub.cs

```diff
@@ -346,60 +394,89 @@ partial class Build
             };
         });
 
-    Target ExtractReleaseNotes => _ => _
-      .Unlisted()
-      .Executes(() =>
-       {
-           Console.WriteLine("Reading changelog...");
-
-           var changelogPath = RootDirectory / "docs" / "CHANGELOG.md";
-           var changelog = File.ReadAllText(changelogPath);
-
-           // find first header
-           var releaseNotesStart = changelog.IndexOf($"## [Release {FullVersion}]");
-           var firstContent = changelog.IndexOf("##", startIndex: releaseNotesStart + 3);
-           var releaseNotesEnd = changelog.IndexOf($"## [Release", firstContent);
-
-           var artifactsLink = Environment.GetEnvironmentVariable("PIPELINE_ARTIFACTS_LINK");
-
-           var sb = new StringBuilder();
-           sb.AppendLine($"⚠ 1. Download the NuGet packages for the release from [this link]({artifactsLink}) and upload to nuget.org");
-           sb.AppendLine("⚠ 2. Download the signed MSI assets and native symbols from GitLab and attach to this release before publishing");
-           sb.AppendLine();
-           sb.Append(changelog, firstContent, releaseNotesEnd - firstContent);
-
-           Console.WriteLine(sb.ToString());
-
-           // need to encode the release notes for use by github actions
-           // see https://trstringer.com/github-actions-multiline-strings/
-           sb.Replace("%","%25");
-           sb.Replace("\n","%0A");
-           sb.Replace("\r","%0D");
-
-           Console.WriteLine("::set-output name=release_notes::" + sb.ToString());
-
-           Console.WriteLine("Release notes generated");
-       });
-
-
     Target DownloadAzurePipelineArtifacts => _ => _
        .Unlisted()
+       .Description("Downloads the latest artifacts from Azure Devops that has the provided version")
        .DependsOn(CreateRequiredDirectories)
        .Requires(() => AzureDevopsToken)
        .Requires(() => Version)
+       .Requires(() => TargetBranch)
        .Executes(async () =>
        {
             // Connect to Azure DevOps Services
             var connection = new VssConnection(
                 new Uri(AzureDevopsOrganisation),
                 new VssBasicCredential(string.Empty, AzureDevopsToken));
 
-            // Get a GitHttpClient to talk to the Git endpoints
+            // Get an Azure devops client
             using var buildHttpClient = connection.GetClient<BuildHttpClient>();
 
-            var branch = $"refs/tags/v{FullVersion}";
+            // Get all the builds to TargetBranch that were triggered by a CI push
+
+            var builds = await buildHttpClient.GetBuildsAsync(
+                             project: AzureDevopsProjectId,
+                             definitions: new[] { AzureDevopsConsolidatePipelineId },
```

Comment:
Is all this below really necessary? You target the consolidated pipeline, so you know you're not targeting no-op so we should have artifacts on the first build, unless it was unsucessful.

Reply:
That's... a good point 🤔 If it's Individual CI, on the master pipeline, it should have a build (successful or not).

Nevertheless, this does still solve a slight hack we had before, where I was selecting a build by start order, _not_ by the actual git commit order. Generally that shouldn't be an issue, but if we _manually_ trigger builds on master (which we sometimes do e.g. trying to bisect and issue, or retrying flaky test for coverage reasons), then we could end up selecting the wrong build.

So in conclusion, yes, this is generally overkill, but is probably still worth keeping now I've written it? 😂

Reply:
Makes complete sense. Can you add a comment then explaining why we're doing it? Because I can already see me stumble on that code in 6 months and say... "Hmm, I will remove that, that's not needed" :)